deploy/transport-controller-deployment.yaml.template

```diff
@@ -0,0 +1,89 @@
+apiVersion: v1
+kind: ConfigMap
+metadata:
+  name: transport-controller-config
+  namespace: ${WDS_NAME}-system
+data:
+  get-kubeconfig.sh: |
+    #!/bin/bash
+    # this script receives a ControlPlane name and a parameter
+    # that determines whether to extract the ControlPlane's in-cluster kubeconfig
+    # or the external kubeconfig (if set to "true", the first will be retrieved).
+    # The function returns the requested kubeconfig's content in base64.
+    # it assumes that it runs within a kubeflex context.
+
+    cpname="$1"
+    get_incluster_key="$2"
+
+    key=""
+    if [[ $get_incluster_key == "true" ]]; then
+      key=$(kubectl get controlplane $cpname -o=jsonpath='{.status.secretRef.inClusterKey}');
+    else
+      key=$(kubectl get controlplane $cpname -o=jsonpath='{.status.secretRef.key}');
+    fi
+
+    # get secret details
+    secret_name=$(kubectl get controlplane $cpname -o=jsonpath='{.status.secretRef.name}')
+    secret_namespace=$(kubectl get controlplane $cpname -o=jsonpath='{.status.secretRef.namespace}')
+
+    # get the kubeconfig in base64
+    kubectl get secret $secret_name -n $secret_namespace -o=jsonpath="{.data.$key}"
+---
+
+apiVersion: apps/v1
+kind: Deployment
+metadata:
+  name: transport-controller
+  namespace: ${WDS_NAME}-system
+spec:
+  replicas: 1
+  selector:
+    matchLabels:
+      name: transport-controller
+  template:
+    metadata:
+      labels:
+        name: transport-controller
+    spec:
+      serviceAccountName: kubestellar-controller-manager
+      initContainers:
+      - name: setup-wds-kubeconfig
+        image: quay.io/kubestellar/kubectl
+        imagePullPolicy: Always
+        command: [ "bin/sh", "-c", "bash /mnt/config/get-kubeconfig.sh ${WDS_NAME} true | base64 -d > /mnt/shared/wds-kubeconfig"]
+        volumeMounts:
+        - name: config-volume
+          mountPath: /mnt/config
+        - name: shared-volume
+          mountPath: /mnt/shared
+      - name: setup-imbs-kubeconfig
+        image: quay.io/kubestellar/kubectl
+        imagePullPolicy: Always
+        command: [ "bin/sh", "-c", "bash /mnt/config/get-kubeconfig.sh ${IMBS_NAME} false | base64 -d > /mnt/shared/transport-kubeconfig"]
+        volumeMounts:
```

Comment:
why the flag for in-cluster is false here ? isn't IMBS also in-cluster ?

Reply:
from transport point of view - no.
transport runs in cluster wds, while imbs is external

Reply:
wds:
transport controller is deployed in the hosting cluster in the wds1-system namespace (since it's per wds).
for this I think we can easily understand we need to use in cluster config.

imbs:
transport controller is running in wds, not in imbs (or ITS). 
therefore, the transport controller is external to imbs.
the transport controller pod is running in a pod in WDS, while imbs is a different space. 
we can even run imbs not through KubeFlex (just a regular kind cluster) and use it the same with the external access.
@MikeSpreitzer maybe you have a better explanation than the above?

Reply:
@nirrozenbaum  I think that both spaces should be IN-Cluster

Reply:
@nirrozenbaum the transport controller is not running in a Pod in WDS as WDS can't run any pod. The TC is running on the hosting cluster and is watching/interacting with both WDS and IMBS .  Both of those spaces/CP are hosted in the same cluster as the TC Pod so you should access the in-cluster. 
When you were running the controller as EXE you were indeed external to the cluster (and then you were external to BOTH spaces) .

Reply:
@pdettori ^^ ??

Reply:
I just validated and ran it locally using incluster for both wds+imbs and it works.
maybe since imbs is vcluster, it also works using the external access.

in general, I think using external access is the "more correct" way, cause this allows us to have it working in any kind of setup, with imbs in KF or without KF, in a pod or in process.
on the other hand, using the incluster is specifically for running everything in KubeFlex. 
it will work in current setup cause we use KF for all.

Reply:
I mean "more correct" in theory. in practice, here we always use KF.

Reply:
@nirrozenbaum remember that although you have expressed a desire to use also an ITS (a.k.a IMBS) as an external cluster you are getting secrets for vcluster that are created in the kubeflex hosting cluster, and kubeflex creates that in-cluster key. I do not know how the external address based on `localtest.me` would work for a process running in a pod because it resolves to 127.0.0.1 and a pod would see that as its local network namespace.

Reply:
my intention was only "conceptually". we can conceptually run ITS in a different cluster and use the external.
but I have no problem changing to incluster, this deployment is used when everything is in KF.
if we want in the future to run ITS in a different cluster outside of KF we will take of that then (anyway current deployment reads from KF, as you mentioned @pdettori).

I will change to use incluster for both.

Reply:
fixed

Reply:
@nirrozenbaum the running in "wds-namespace-system"  is not relevant at all here. We have a mix-up in the definitions. 
You have an api-server that is exposed as the WDS space. When you deploy resources into WDS you basically interact with that api-server and the etcd server it uses. 
Your Transport Controller is NOT deployed to WDS at all , it deployed to a regular cluster (the hosting cluster in your case) , you supply that controller with config information to access other "clusters"  in your case the api-servers of WDS and IMBS.  
Because all the Pods involved - the transport controller the WDS-api-server and the IMBS-api-server  reside ion the same cluster you can and should use the in-cluster access. If you use the external access the Pod won't be able to resolve the cluster IP.
We need to be carful when we say "run in a cluster"  VS "interact with cluster" .

Reply:
@pdettori It reminds me that I think the KS controller has a bug on how it test if to access in-cluster Vs external.  I think that the current code will choose "in-cluster"  as long as the controller is running in a cluster (even if that cluster is different than that of KF). I guess the assumption is that KS controller pod and KF both run in the same cluster